pkg/infra/usagestats/service/api.go

```diff
@@ -0,0 +1,31 @@
+package service
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+const rootUrl = "/api/admin"
+
+func (uss *UsageStats) registerAPIEndpoints() {
+	uss.RouteRegister.Group(rootUrl, func(subrouter routing.RouteRegister) {
+		subrouter.Get("/usage-report-preview", middleware.ReqGrafanaAdmin, routing.Wrap(uss.getUsageReportPreview))
+	})
+}
+
+func (uss *UsageStats) getUsageReportPreview(ctx *models.ReqContext) response.Response {
+	if !uss.Cfg.ReportingEnabled {
+		return response.JSON(http.StatusUnauthorized, "Reporting is not enabled")
```

Comment:
we might want to think about this endpoint still being available even if the actual sending is disabled, so users can review it to verify before enabling if they're concerned about privacy

Reply:
That's a good point, will implement